spec/timezone.html

```diff
@@ -657,11 +647,29 @@ <h1>
         <dd>
           This operation is the internal implementation of the `Temporal.TimeZone.prototype.getOffsetStringFor` method.
           If the given _timeZone_ is an Object, it observably calls _timeZone_'s `getOffsetNanosecondsFor` method.
+          If the offset represents an integer number of minutes, then the output will be formatted like ±HH:MM.
+          Otherwise, the output will be formatted like ±HH:MM:SS or (if the offset does not evenly divide into seconds) ±HH:MM:SS.fff… where the "fff" part is a sequence of at least 1 and at most 9 fractional seconds digits with no trailing zeroes.
         </dd>
       </dl>
       <emu-alg>
         1. Let _offsetNanoseconds_ be ? GetOffsetNanosecondsFor(_timeZone_, _instant_).
-        1. Return FormatOffsetTimeZoneIdentifier(_offsetNanoseconds_, ~separated~).
+        1. Let _offsetMinutes_ be truncate(_offsetNanoseconds_ / (6 × 10<sup>10</sup>)).
+        1. Let _offsetString_ be FormatOffsetTimeZoneIdentifier(_offsetMinutes_ × (6 × 10<sup>10</sup>), ~separate~).
+        1. Let _subMinuteNanoseconds_ be abs(_offsetNanoseconds_) modulo (6 × 10<sup>10</sup>).
+        1. If _subMinuteNanoseconds_ = 0, return _offsetString_.
+        1. If _offsetMinutes_ = 0 and _offsetNanoseconds_ &lt; 0, let _offsetString_ be *"-00:00"*.
+        1. Let _seconds_ be floor(_subMinuteNanoseconds_ / 10<sup>9</sup>) modulo 60.
+        1. Let _s_ be ToZeroPaddedDecimalString(_seconds_, 2).
+        1. Let _nanoseconds_ be _subMinuteNanoseconds_ modulo 10<sup>9</sup>.
+        1. If _nanoseconds_ &ne; 0, then
+          1. Let _fraction_ be ToZeroPaddedDecimalString(_nanoseconds_, 9).
+          1. Set _fraction_ to the longest possible substring of _fraction_ starting at position 0 and not ending with the code unit 0x0030 (DIGIT ZERO).
+          1. Let _post_ be the string-concatenation of the code unit 0x003A (COLON), _s_, the code unit 0x002E (FULL STOP), and _fraction_.
+        1. Else if seconds &ne; 0, then
+          1. Let _post_ be the string-concatenation of the code unit 0x003A (COLON) and _s_.
+        1. Else,
+          1. Let _post_ be the empty String.
+        1. Return the string-concatenation of _offsetString_ and _post_.
```

Comment:
Isn't the final Else branch impossible to reach?
```suggestion
          1. Let _suffix_ be the string-concatenation of the code unit 0x002E (FULL STOP) and _fraction_.
        1. Else,
          1. Assert: _seconds_ &ne; 0.
          1. Let _suffix_ be *""*.
        1. Return the string-concatenation of _offsetString_, the code unit 0x003A (COLON), _s_, and _suffix_.
```

Reply:
Good catch, yes it's now unreachable. Will fix.

Reply:
Since we're refactoring this AO anyways, I made a few more editorial changes that IMO better communicate the cascade of early returns if seconds are zero, and if not then if nanos are zero, and finally to fall through to the nano case.

```
1. Let _offsetNanoseconds_ be ? GetOffsetNanosecondsFor(_timeZone_, _instant_).
1. Let _offsetMinutes_ be truncate(_offsetNanoseconds_ / (6 × 10<sup>10</sup>)).
1. Let _offsetString_ be FormatOffsetTimeZoneIdentifier(_offsetMinutes_ × (6 × 10<sup>10</sup>), ~separate~).
1. Let _subMinuteNanoseconds_ be abs(_offsetNanoseconds_) modulo (6 × 10<sup>10</sup>).
1. If _subMinuteNanoseconds_ = 0, then
  1. Return _offsetString_.
1. If _offsetMinutes_ = 0 and _offsetNanoseconds_ &lt; 0, set _offsetString_ to *"-00:00"*.
1. Let _seconds_ be floor(_subMinuteNanoseconds_ / 10<sup>9</sup>) modulo 60.
1. Let _ss_ be ToZeroPaddedDecimalString(_seconds_, 2).
1. Let _nanoseconds_ be _subMinuteNanoseconds_ modulo 10<sup>9</sup>.
1. If _nanoseconds_ = 0, then
  1. Return the string-concatenation of _offsetString_, the code unit 0x003A (COLON), and _ss_.
1. Let _fraction_ be ToZeroPaddedDecimalString(_nanoseconds_, 9).
1. Set _fraction_ to the longest prefix of _fraction_ ending with a code unit other than 0x0030 (DIGIT ZERO).
1. Return the string-concatenation of _offsetString_, the code unit 0x003A (COLON), _ss_, the code unit 0x002E (FULL STOP), and _fraction_.
```